### PR TITLE
PLANET-7597 Fix Block Usage Report 'group by post id' links'

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-7597](https://jira.greenpeace.org/browse/PLANET-7597)

If this implementation was working in the plugin, it's weird that it no longer works in the theme. Maybe WordPress changed something in `WP_List_Table`.

**Note:** this is a bit of a messy fix, so any feedback is welcome!

### Testing

- In the [broken version](https://www-dev.greenpeace.org/test-pluto/wp-admin/admin.php?page=plugin_blocks_report&group=post_id), there is no link in the "title" column and the row actions links are not the page edit url, they just reload the page:

Screenshot 2024-10-01 at 16 09 28

- In the [fixed version](https://www-dev.greenpeace.org/test-nix/wp-admin/admin.php?page=plugin_blocks_report&group=post_id), both of these issues are no longer present:

Screenshot 2024-10-01 at 16 09 41

### CODE REVIEW

**Comment by mardelnet** [comment]
@mleray would you mind adding a step-by-step testing guide?

**Review by mardelnet — COMMENTED** [review]
Good job @mleray 
I have a couple of comments:

1. The "Pattern Report" page (`/wp-admin/admin.php?page=plugin_patterns_report`) is broken since `get_post_actions` is missing the fourth parameter.

2. The link to the front is adding an extra "wp-admin" to the URL:
Screenshot-from-2024-10-02-09-29-05

File: src/BlockReportSearch/RowActions.php [hunk 1]
```diff
@@ -14,25 +14,30 @@ class RowActions
     /**
      * Add action links to a row
      *
-     * @param array  $item        Item.
-     * @param string $column_name Current column name.
-     * @param string $primary     Primary column name.
+     * @param array  $item              Item.
+     * @param string $column_name       Current column name.
+     * @param string $primary           Primary column name.
+     * @param string $potential_post_id Primary column name.
```
[the remark this answers is not in the document]
Comment (mleray) [reply src/BlockReportSearch/RowActions.php]: Ooops 😬 It's removed now anyway!

File: src/BlockReportSearch/RowActions.php [hunk 2]
```diff
@@ -14,25 +14,30 @@ class RowActions
...
+        string $column_name,
+        string $primary,
+        string $potential_post_id
+    ): array {
         if ($column_name !== $primary) {
             return [];
         }
 
-        $id = (int) $item['post_id'];
+        $id = empty($item['post_id']) ? (int) $potential_post_id : (int) $item['post_id'];
```
[the remark this answers is not in the document]
Comment (mleray) [reply src/BlockReportSearch/RowActions.php]: I think it's more intuitive than using the `latest_row` variable, so I've updated the code!

**Review by mardelnet — COMMENTED** [review]
I see a fatal error in the " Pattern Report" page now.
Can you check it, @mleray?

**Comment by mleray** [comment]
> I see a fatal error in the " Pattern Report" page now. Can you check it, @mleray?

@mardelnet I don't see a fatal error 🤔 But apparently we don't have the guid param there so we might need to use latest_row after all... Sorry about that, I should have tested better!

**Comment by mleray** [comment]
@mardelnet I believe it's good now 🤞 I know that SonarCloud is complaining about duplicated code, maybe we could open a follow-up ticket to see if we could refactor this?

File: src/BlockReportSearch/RowActions.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mleray) [reply src/BlockReportSearch/RowActions.php]: Well it's back, but I've updated the comment!

**Review by mardelnet — APPROVED** [review]

**Pull request merged**